packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -585,7 +585,7 @@ export default class TradeStore extends BaseStore {
                             });
                         }
                     } else if (response.buy) {
-                        if (this.proposal_info[type]?.id !== proposal_id) {
+                        if (this.proposal_info[type] && this.proposal_info[type].id !== proposal_id) {
```

Comment:
This change won't make any difference 🤔

Reply:
For this one, in the first case, if the value is undefined, it will compare it to the right side and return true, for the second case, it will not do comparison unless the value is defined. This causes the error to throw every time, the value is not yet resolved.